horizon/main.go

```diff
@@ -0,0 +1,113 @@
+package horizon
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/stellar/go-stellar-base/horizon/structs"
+)
+
+const TIMEOUT = 30 * time.Second
```

Comment:
The main reason I added this timeout in the first place is that submit transaction request requires more time (when sending many transactions and horizon is ordering them). Ideally, we should have two timeouts: for normal GET requests and for submitting transactions.

So maybe we should have two constants (I will explain why below): `DEFAULT_REQUEST_TIMEOUT` and `DEFAULT_SUBMIT_REQUEST_TIMEOUT`? Then if user wants to overwrite it, it can be setting `Client` struct's `RequestTimeout` and `SubmitRequestTimeout` fields. On the other hand, this can be a bit overcomplicating. What do you think?